data/transactions/logic/eval.go

```diff
@@ -899,12 +899,16 @@ func (cx *EvalContext) remainingInners() int {
 
 func (cx *EvalContext) step() error {
 	opcode := cx.program[cx.pc]
-	spec := &opsByOpcode[cx.version][opcode]
+	spec, _ := getLeaf(cx.program, cx.pc, cx.version, &opsByOpcode[cx.version][opcode])
```

Comment:
Since this is the hot loop, we may need to take care and do something funky to make the fast path fast. For example, we might do the simple lookup, and only call `getLeaf` if we see it's a `multi`, because that check can be done without function call overhead.  We'll have to benchmark. (The point is that we don't mind if the multibyte ops at 5% slower or something like that, but we can't slow down all the other one byte ops too.)

Reply:
Ya for some reason I was doing this logic for the assembler instead of the evaluator; I've made the recommended change here and also change how we check deprecated specs.